### PR TITLE
Add database migrations to deploy and development

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-594

### CODE REVIEW

**Comment by MartinEden** [comment]
I haven't found a way of auto-running the documentation generator and then committing that back to the repository as part of TeamCity. So at the moment when you write a new migration you need to run it on your local machine and then commit the changes to the documentation. Other than that, this all working pretty well.

File: migrations/README.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+migrations that get from an empty database to one that has essential data in it
+(like roles and permissions, and the various enum types). These will produce
+identical data to that which is in production, but we will never actually run
+them on production.
+
+We can manage this with Flyway's baseline feature. Once I have written this
+initial suite of migrations we can run:
+
+```
+./flyway.sh baseline -baselineVersion=2017.09.06.1055
```
Comment (richfitz) [inline migrations/README.md]: this should say run the container

File: migrations/flyway.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+docker build --tag montagu-migrations .
```
Comment (richfitz) [inline migrations/flyway.sh]: this file can be removed

File: migrations/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  && mv flyway-${FLYWAY_VERSION}/* . \
+  && rm flyway-commandline-${FLYWAY_VERSION}.tar.gz \
+  && sed -i 's/bash/sh/' /flyway/flyway \
+  && ln -s /flyway/flyway /usr/local/bin/flyway
+
+COPY migrations/sql /sql
+COPY migrations/flyway.conf /flyway/conf/flyway.conf
+
+ENTRYPOINT ["flyway"]
+CMD ["migrate", "-baselineOnMigrate=true", "-baselineVersion=0"]
```
Comment (richfitz) [inline migrations/Dockerfile]: CMD should just be `"migrate"`

**Pull request merged**